manticore/ethereum/__init__.py

```diff
@@ -325,16 +325,17 @@ def _run_solc(source_file, solc_bin=None, solc_remaps=[]):
         #solc path search is a mess #fixme
         #https://solidity.readthedocs.io/en/latest/layout-of-source-files.html
         current_folder = os.getcwd()
-        abs_filename = os.path.abspath(source_file.name)
-        working_folder, filename = os.path.split(abs_filename)
+        filepath = os.path.abspath(source_file.name)
+
+        filename = filepath[filepath.rindex(os.path.sep)+1:]
```

Comment:
The `filename` is used below  to detect `solc` run within `snap`:

https://github.com/trailofbits/manticore/pull/1171/files#diff-652275d107ee027643e6dc52afea9d2eR343

Reply:
Hmm... isn't it fine inside `[]`?

```
$ git grep "[[].*\s*[+]1:"
manticore/ethereum/__init__.py:        filename = filepath[filepath.rindex(os.path.sep)+1:]
tests/test_binaries.py:        filename = filename[len(os.getcwd())+1:]
tests/test_binaries.py:        filename = filename[len(os.getcwd())+1:]
tests/test_binaries.py:        filename = filename[len(os.getcwd())+1:]
tests/test_memory.py:        self.assertItemsEqual(mem[addr+1:addr+7], b'bcdabc')
```

it seems that we don't use spaces in similar cases.